### PR TITLE
Sync: Handle unknown block root event from regular sync

### LINKED ISSUE #1346: Backward sync controlled by sync
**Describe the bug**
Currently we abuse selectively emiting "unknownBlockRoot" in the block pool to suppress backwards sync. This is confusing, since the block pool should not be concerned with syncing strategy. There might also be eventually other consumers of "unknownBlockRoot" which need every unknown block root to fire an event.

**Expected behavior**
Sync controls backwards sync.

**Screenshots**
https://github.com/ChainSafe/lodestar/blob/master/packages/lodestar/src/chain/blocks/pool.ts#L35

### PR DESCRIPTION
**Pull request by twoeths**
resolves #1346 

+ This is based on the new block pool work
+ Update sync status to` SYNCED` on `syncCompleted` event
+ Sync: Separate `getUnknownRootPeers` and `getSyncPeer`

### CODE REVIEW

File: packages/lodestar/src/chain/emitter.ts [hunk 1]
```diff
@@ -40,6 +39,8 @@ export interface IChainEvents {
   "forkChoice:reorg": (head: IBlockSummary, oldHead: IBlockSummary, depth: number) => void;
   "forkChoice:justified": (checkpoint: Checkpoint) => void;
   "forkChoice:finalized": (checkpoint: Checkpoint) => void;
+
+  "block:unknownRoot": (root: Root) => void;
```
Comment (mpetrunic) [inline packages/lodestar/src/chain/emitter.ts]: isn't that present in error:block?

Comment (twoeths) [reply packages/lodestar/src/chain/emitter.ts]: when doing block `validate` we throw `error:block` with `ERR_PARENT_UNKNOWN` code and chain handle that, then chain puts to pending block pool and emits `block:unknownRoot` with missing ancestor root then sync will handle that.
So I'd separate `error:block` and `block:unknownRoot` event.

Comment (mpetrunic) [reply packages/lodestar/src/chain/emitter.ts]: hmm but do we need to fire block:unknownRoot from pool again, I mean everyone can listen to error:block and check if code is missing parent, right?

Comment (twoeths) [reply packages/lodestar/src/chain/emitter.ts]: I thought about it, my understanding is `BlockPool` merely a data storage, chain is the controller so I did not fire `block:unknownRoot` inside block pool and let chain does the job.
cc @wemeetagain to see if this is the correct way

**Comment by twoeths** [comment]
Update: in regular sync, we want to handle ERR_BLOCK_IS_ALREADY_KNOWN because it maybe auto-processed by pending block pool already (previously we still emit "block" event in `validate.ts` but now we switch to ERR_BLOCK_IS_ALREADY_KNOWN block error code)

**Review by wemeetagain — APPROVED** [review]

**Pull request merged**